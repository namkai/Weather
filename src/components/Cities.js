import PropTypes from 'prop-types';
import React from 'react';
import City from './City';

const Cities = ({ storedCities, fetchWeather }) => {
	const cityList = storedCities.map(city => <City {...city} fetchWeather={fetchWeather}/>);
	return (
		<ul className="nav nav-pills nav-stacked flex-column">
			<li className="nav-header">Recent Cities</li>
			{cityList}
		</ul>
	);
};

Cities.defaultProps = {
	storedCities: [],
	fetchWeather: PropTypes.func,
};

Cities.propTypes = {
	storedCities: PropTypes.arrayOf(PropTypes.object),
	fetchWeather: PropTypes.func,
};

export default Cities;