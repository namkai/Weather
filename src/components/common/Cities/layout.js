import React from 'react';

const CitiesLayout = ({ children, clearStoredCities }) => (
  <ul className="nav nav-pills nav-stacked flex-column">
    <li className="nav-header">Recent Cities
      <small> Â· <a href="#" onClick={clearStoredCities}>Clear</a></small>
    </li>
    {children}
  </ul>
);

export default CitiesLayout;
