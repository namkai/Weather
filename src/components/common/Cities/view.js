import PropTypes from 'prop-types';
import React from 'react';
import CitiesLayout from './layout';
import City from '../City';

const Cities = ({ storedCities, fetchWeather, clearStoredCities }) => {
  let view = null;
  if (storedCities.length > 0) {
    view = storedCities.map((city, idx) => <City name={city} key={idx} fetchWeather={fetchWeather} />);
  }
  return <CitiesLayout clearStoredCities={clearStoredCities}>{ view }</CitiesLayout>;
};

Cities.defaultProps = {
  storedCities: [],
  fetchWeather: PropTypes.func,
};

Cities.propTypes = {
  storedCities: PropTypes.arrayOf(PropTypes.string),
  fetchWeather: PropTypes.func,
};

export default Cities;
