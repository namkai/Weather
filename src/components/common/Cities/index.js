import { compose, lifecycle, pure } from 'recompose';

import Cities from './view';

const lifecycleMethods = lifecycle({
  componentWillMount() {
    let cities = localStorage.getItem('storedCities');
    if (cities !== null && cities !== '') {
      cities = cities.split('&');
      this.props.updateStoredCities(cities);
    }
  },
  componentDidUpdate() {
    const stringCities = this.props.storedCities.join('&');
    localStorage.setItem('storedCities', stringCities);
  },
});

const PureCities = pure(Cities);

export default compose(lifecycleMethods)(PureCities);