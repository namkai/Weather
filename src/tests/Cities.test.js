import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Cities from '../components/Cities';

it('renders correctly', () => {
	const cities = shallow(
		<Cities />
	);
	expect(toJson(cities)).toMatchSnapshot();
});