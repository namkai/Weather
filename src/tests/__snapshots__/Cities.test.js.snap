// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="nav nav-pills nav-stacked flex-column"
>
  <li
    className="nav-header"
  >
    Recent Cities
  </li>
</ul>
`;
